SlingModelPersist/src/main/java/org/apache/sling/models/persistor/impl/ModelPersistorImpl.java

```diff
@@ -129,8 +130,8 @@ public void persist(final @NotNull String nodePath, final @NotNull Object instan
             } else {
                 Resource r = resource;
                 fields.stream()
-                        .filter(ReflectionUtils::isNotTransient)
-                        .filter(ReflectionUtils::isSupportedType)
+                        .filter(field -> ReflectionUtils.isNotTransient(field, isUpdate))
+                        .filter(field -> ReflectionUtils.isSupportedType(field) || ReflectionUtils.isCollectionOfPrimitiveType(field))
```

Comment:
It would seem like a good idea to have some logging as to why a field is not being persisted.